react-components/src/components/RevealToolbar/SettingsContainer/HighFidelityContainer.tsx

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { Menu } from '@cognite/cogs.js';
+import { type ReactElement, useEffect, useMemo } from 'react';
+import { useReveal } from '../../RevealContainer/RevealContext';
+import { type ResolutionOptions } from '@cognite/reveal';
+import { type HighFidelityProps } from './types';
+
+export const HighFidelityContainer = ({
+  isHighFidelityMode,
+  setHighFidelityMode
+}: HighFidelityProps): ReactElement => {
+  const viewer = useReveal();
+
+  const defaultsQualityConfig = useMemo(() => {
+    return {
+      cadBudget: { ...viewer.cadBudget },
+      pointCloudBudget: { ...viewer.pointCloudBudget },
+      resolutionOptions: {
+        maxRenderResolution: 1.4e6,
+        movingCameraResolutionFactor: 1
+      } satisfies ResolutionOptions
+    };
+  }, [viewer]);
+
+  const qualityConfig = useMemo(() => {
+    if (isHighFidelityMode) {
+      return {
+        pointCloudBudget: {
+          numberOfPoints: 3 * defaultsQualityConfig.pointCloudBudget.numberOfPoints
+        },
+        cadBudget: {
+          maximumRenderCost: 3 * defaultsQualityConfig.cadBudget.maximumRenderCost,
+          highDetailProximityThreshold: defaultsQualityConfig.cadBudget.highDetailProximityThreshold
+        },
+        resolutionOptions: {
+          maxRenderResolution: Infinity
+        }
```

Comment:
Would be nice to be able for the user to optionally pass the high fidelity settings into the component, and then have this be the default.